diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -411,6 +454,71 @@ namespace diff_drive_controller{
       left_wheel_joints_[i].setCommand(vel_left);
       right_wheel_joints_[i].setCommand(vel_right);
     }
+
+    // Publish wheel data:
+    if (publish_joint_trajectory_controller_state_ && joint_trajectory_controller_state_pub_->trylock())
+    {
+      joint_trajectory_controller_state_pub_->msg_.header.stamp = time;
+
+      for (size_t i = 0; i < wheel_joints_size_; ++i)
+      {
+        const double control_duration = (time - time_previous_).toSec();
+
+        const double left_wheel_acc = (left_wheel_joints_[i].getVelocity() - vel_left_previous_[i]) / control_duration;
+        const double right_wheel_acc = (right_wheel_joints_[i].getVelocity() - vel_right_previous_[i]) / control_duration;
+
+        // Actual
+        joint_trajectory_controller_state_pub_->msg_.actual.positions[i]     = left_wheel_joints_[i].getPosition();
+        joint_trajectory_controller_state_pub_->msg_.actual.velocities[i]    = left_wheel_joints_[i].getVelocity();
+        joint_trajectory_controller_state_pub_->msg_.actual.accelerations[i] = left_wheel_acc;
+        joint_trajectory_controller_state_pub_->msg_.actual.effort[i]        = left_wheel_joints_[i].getEffort();
+
+        joint_trajectory_controller_state_pub_->msg_.actual.positions[i + wheel_joints_size_]     = right_wheel_joints_[i].getPosition();
+        joint_trajectory_controller_state_pub_->msg_.actual.velocities[i + wheel_joints_size_]    = right_wheel_joints_[i].getVelocity();
+        joint_trajectory_controller_state_pub_->msg_.actual.accelerations[i + wheel_joints_size_] = right_wheel_acc;
+        joint_trajectory_controller_state_pub_->msg_.actual.effort[i+ wheel_joints_size_]         = right_wheel_joints_[i].getEffort();
+
+        // Desired
+        joint_trajectory_controller_state_pub_->msg_.desired.positions[i]    += vel_left_desired * cmd_dt;
+        joint_trajectory_controller_state_pub_->msg_.desired.velocities[i]    = vel_left_desired;
```

Comment:
What is your desired velocity? Is it the one set directly by `cmd_vel` or after applying the limiters? 

To me an internal controller state monitor would use the current value to fill `actual` and the one you are sending out in this loop on the joint interfaces as `desired`.

Reply:
Here `desired` as been filled with the `cmd_vel` prior to limitations, while `actual` is the one reported by (from) the `JointHandle`.